src/ast/passes/semantic_analyser.cpp

```diff
@@ -3987,6 +4006,30 @@ bool SemanticAnalyser::update_string_size(SizedType &type,
   return false;
 }
 
+SizedType SemanticAnalyser::create_merged_tuple(const SizedType &left,
+                                                const SizedType &right)
+{
+  assert(left.IsTupleTy() && right.IsTupleTy() &&
+         (left.GetFieldCount() == right.GetFieldCount()));
+
+  SizedType type;
+  std::vector<SizedType> new_elems;
+  for (ssize_t i = 0; i < left.GetFieldCount(); i++) {
+    const auto &leftTy = left.GetField(i).type;
+    const auto &rightTy = right.GetField(i).type;
+
+    assert(leftTy.GetTy() == rightTy.GetTy());
+    if (leftTy.IsTupleTy()) {
+      new_elems.push_back(create_merged_tuple(leftTy, rightTy));
+    } else {
+      new_elems.push_back(leftTy.GetSize() > rightTy.GetSize() ? leftTy
+                                                               : rightTy);
+    }
+  }
+  type = CreateTuple(bpftrace_.structs.AddTuple(new_elems));
+  return type;
```

Comment:
Nit, you can probably just do
```suggestion
  return CreateTuple(bpftrace_.structs.AddTuple(new_elems));
```
and drop the `SizedType type;` decl above the loop.

Just cleaner code.